### PR TITLE
Bug fixes on Statistics Plugin

### PR DESCRIPTION
**Pull request by amicis31**
The work done is:
- Fix download csv (statistics plugin) to make it work on Firefox.
- Fix the submission link when showing table in statistics plugin,

### CODE REVIEW

**Review by crdgonzalezca — CHANGES_REQUESTED** [review]
@amicis31 , Can you add a description to this PR? Telling what it is fixing as I don't know. Once it is done, I can start reviewing it. Thanks :)

**Comment by amicis31** [comment]
@crdgonzalezca ok I hope is clear. :smile:

File: inginious/frontend/plugins/statistics/pages/course_admin_statistics.html [hunk 1]
```diff
@@ -77,4 +77,17 @@ <h4>Submissions Verdict (BEST)</h4>
...
+        element = download_csv_buttons[it];
+        fun_name = element.getAttribute('onclick');
+
+        // windown['function']. To use check function from a string.
+        if( window[fun_name] !== "function" ) {
+            element.style.visibility = "hidden";
+        }
+    }
+    
 </script>
```
Comment (crdgonzalezca) [inline inginious/frontend/plugins/statistics/pages/course_admin_statistics.html]: I was testing this but the link was hidden but I had some submissions, so I guess I should be shown to me and I can download the file. So, I do not understand why we are doing this or what are the cases we should hide the option. Can you clarify this o fix it?

File: inginious/frontend/plugins/statistics/pages/user_statistics.html [hunk 2]
```diff
@@ -21,12 +21,24 @@ <h3>Bar Submissions Per Tasks</h3>
...
+
+    // Checking if csv download function exist in the code
+    var download_csv_buttons = document.getElementsByClassName('download_csv');
+    for(let it = 0; it < download_csv_buttons.length; it++){
+        element = download_csv_buttons[it];
+        fun_name = element.getAttribute('onclick');
+
+        // windown['function']. To use check function from a string.
+        if( window[fun_name] !== "function" ) {
+            element.style.visibility = "hidden";
```
Comment (crdgonzalezca) [inline inginious/frontend/plugins/statistics/pages/user_statistics.html]: The same here, I do not understand very well when this should work as this was hidden to me testing this in my statistics and course statistics. Maybe we should show the option when there's not data in the statistics. But please, comment the use cases for this.

**Review by crdgonzalezca — COMMENTED** [review]
@amicis31 please see the comments below as there are some things I don't understand yet.

File: inginious/frontend/plugins/statistics/pages/user_statistics.html (hunk 2 above)
Comment (amicis31) [reply inginious/frontend/plugins/statistics/pages/user_statistics.html]: This piece of code checks if there exists a function with the name fun_name. In this case is the function that is suppose to run after the click of the link.

So at the end the function checks that this link can run a js function. And shows it or hides it, depending of the case

File: inginious/frontend/plugins/statistics/pages/course_admin_statistics.html (hunk 1 above)
Comment (amicis31) [reply inginious/frontend/plugins/statistics/pages/course_admin_statistics.html]: This check if the function is declared. This is to solve "Download CSV option does not work, if it is not available, it shouldn't be shown to the user" issue.

**Review by crdgonzalezca — APPROVED** [review]
Approved. I'll merge it. :smile:

**Pull request merged**